kani-compiler/src/kani_middle/transform/check_uninit/uninit_visitor.rs

```diff
@@ -0,0 +1,812 @@
+                                        value: true,
+                                        position: InsertPosition::After,
+                                    })
+                                }
+                                intrinsic => {
+                                    self.push_target(MemoryInitOp::Unsupported {
+                                    reason: format!("Kani does not support reasoning about memory initialization of intrinsic `{intrinsic}`."),
+                                });
+                                }
+                            }
+                        }
+                        InstanceKind::Item => {
+                            if instance.is_foreign_item() {
+                                match instance.name().as_str() {
+                                    "alloc::alloc::__rust_alloc"
+                                    | "alloc::alloc::__rust_realloc" => {
+                                        /* Memory is uninitialized, nothing to do here. */
+                                    }
+                                    "alloc::alloc::__rust_alloc_zeroed" => {
+                                        /* Memory is initialized here, need to update shadow memory. */
+                                        self.push_target(MemoryInitOp::Set {
+                                            operand: Operand::Copy(destination.clone()),
+                                            count: args[0].clone(),
+                                            value: true,
+                                            position: InsertPosition::After,
+                                        });
+                                    }
+                                    "alloc::alloc::__rust_dealloc" => {
+                                        /* Memory is uninitialized here, need to update shadow memory. */
+                                        self.push_target(MemoryInitOp::Set {
+                                            operand: args[0].clone(),
+                                            count: args[1].clone(),
+                                            value: false,
+                                            position: InsertPosition::After,
+                                        });
+                                    }
+                                    _ => {}
+                                }
+                            }
+                        }
+                        _ => {}
+                    }
+                }
+                TerminatorKind::Drop { place, .. } => {
+                    self.super_terminator(term, location);
+                    let place_ty = place.ty(&self.locals).unwrap();
+                    // When drop is codegen'ed, a reference is taken to the place which is later implicitly coerced to a pointer.
+                    // Hence, we need to bless this pointer as initialized.
+                    self.push_target(MemoryInitOp::SetRef {
```

Comment:
`SetRef`is actually a `Set` that applies to the pointer to the place, not the place itself. This is needed because when drop is codegen'd, a reference to the place is coerced under the hood to a pointer, which, without this blessing, will fail a check assertion upon access.

Reply:
Renamed `Get` to `Check` though

Reply:
I'm still confused about this one. Maybe we can check offline. You just added this comment that makes it sound like a check to me:
> ```rust
>    /// Check memory initialization of data bytes in a memory region starting from the reference to
>    /// `operand` and of length `count * sizeof(operand)` bytes.
> ```

Reply:
My bad, the comment is incorrect. "Check" here should mean "Set". Fixed it in the last commit.